### PR TITLE
Don't call isSome and .Value

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

File: src/fsharp/TypeChecker.fs [hunk 1]
```diff
@@ -6821,8 +6821,8 @@ and TcRecdExpr cenv overallTy env tpenv (inherits, optOrigExpr, flds, mWholeExpr
             None,tpenv
 
     let expr,tpenv = 
-        let fldsList = fldsList |> List.choose (fun (n, v) -> if v.IsSome then Some (n, v.Value) else None)
-        TcRecordConstruction cenv overallTy env tpenv optOrigExpr  overallTy fldsList mWholeExpr
+        let fldsList = fldsList |> List.choose (fun (n, v) -> match v with Some v -> Some (n, v) | _ -> None)
```
Comment (0x53A) [inline src/fsharp/TypeChecker.fs]: Option.map?

```F#
v |> Option.map (fun _ -> (n, v))
```

(feel free to ignore this =) )

Comment (forki) [inline src/fsharp/TypeChecker.fs]: send your own PR ;-)

Comment (vasily-kirichenko) [inline src/fsharp/TypeChecker.fs]:
```fsharp
let fldsList = 
  [ for n, v in fldsList do
      match v with 
      | Some v -> yield n, v
      | None -> () ]
```

Comment (forki) [inline src/fsharp/TypeChecker.fs]: send your own PR ;-)

Comment (0x53A) [inline src/fsharp/TypeChecker.fs]: Nah, too much effort :D I'm just stalking github on my coffee break.
And I think I like the version by @vasily-kirichenko better than my own anyway, List Expressions are great.

Comment (vasily-kirichenko) [inline src/fsharp/TypeChecker.fs]: Yes, comprehensions are one of the F# pears and should be used more widely.

Comment (forki) [inline src/fsharp/TypeChecker.fs]: Ok ok let me update the PR

Comment (vasily-kirichenko) [inline src/fsharp/TypeChecker.fs]: I'm not saying that the list expression is what we need here ;)

Comment (matthid) [inline src/fsharp/TypeChecker.fs]: `fldsList |> List.choose (function (n, Some v) -> Some (n, v) | _ -> None)`
